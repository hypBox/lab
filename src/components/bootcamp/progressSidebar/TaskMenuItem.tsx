import React, { FunctionComponent } from "react"
import { createStyles, makeStyles } from "../../material/styles"
import {
  ListItem,
  ListItemText,
  ListItemIcon,
  ListItemSecondaryAction,
  Typography,
} from "../../material/core"
import { FAIcon } from "../../material/icons"
import { ITask } from "../../../types"
import { getTaskIcon, GetTaskIconParams } from "./utility"

export interface TaskMenuItemProps {
  task: ITask
  onclick: CallableFunction
  selectedTask: ITask
}

const useStyles = makeStyles((theme) =>
  createStyles({
    nested: {
      paddingLeft: theme.spacing(4),
    },
  })
)

const Task: FunctionComponent<TaskMenuItemProps> = ({
  task,
  onclick,
  selectedTask,
}) => {
  const classes = useStyles()
  return (
    <ListItem
      button
      key={task.id}
      className={classes.nested}
      selected={selectedTask.id === task.id}
      onClick={() => onclick(task)}
    >
      <ListItemIcon>
        <FAIcon icon={getTaskIcon(task as GetTaskIconParams)} />
      </ListItemIcon>
      <ListItemText
        primary={task.title}
        primaryTypographyProps={{
          variant: "body2",
          color: "textPrimary",
        }}
      />
      <ListItemSecondaryAction>
        <Typography color="textPrimary" variant={"caption"}>
          {task.length}
        </Typography>
      </ListItemSecondaryAction>
    </ListItem>
  )
}

export default Task
