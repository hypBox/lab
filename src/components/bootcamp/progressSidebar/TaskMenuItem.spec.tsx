import React from "react"
import renderer from "react-test-renderer"

import TaskMenuItem from "./TaskMenuItem"
import { TaskProps } from "./TaskMenuItem"

describe("Task Component", () => {
  it("renders as expected", () => {
    const task = {
      id: "t1",
      progress: 80,
      title: "TASK 1",
      type: "task",
      length: "LENGTH",
    }

    const taskProps: TaskProps = {
      onclick: jest.fn(),
      selectedTask: task,
      task,
    }

    const tree = renderer.create(<TaskMenuItem {...taskProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
