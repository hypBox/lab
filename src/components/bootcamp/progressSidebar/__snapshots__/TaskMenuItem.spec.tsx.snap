// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Component renders as expected 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root makeStyles-nested-1 MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction Mui-selected"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-right fa-w-10 "
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
      >
        TASK 1
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <span
      className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
    >
      LENGTH
    </span>
  </div>
</li>
`;
